### PR TITLE
fix: allow assert values command to check partial columns

### LINKED ISSUE #6056: KsqlTester should support partial column matches
The following test should succeed:
```sql
----------------------------------------------------------------------------------------------------
--@test: basic test partial columns
----------------------------------------------------------------------------------------------------
CREATE STREAM foo (id INT KEY, col1 INT, col2 INT) WITH (kafka_topic='foo', value_format='JSON');
CREATE STREAM bar AS SELECT * FROM foo;

INSERT INTO foo (rowtime, id, col1, col2) VALUES (1, 1, 1, 1);
ASSERT VALUES bar (rowtime, id, col1) VALUES (1, 1, 1);
```

But does not because 
```
io.confluent.ksql.test.KsqlTestException: Test failure for assert `ASSERT VALUES BAR (ROWTIME, ID, COL1) VALUES (1, 1, 1)` (Line: 21, Col: 15):
	Expected record does not match actual. Expected: KsqlGenericRecord{key=Struct{ID=1}, value=[ 1 | null ], ts=1} vs. Actual: KsqlGenericRecord{key=Struct{ID=1}, value=[ 1 | 1 ], ts=1}
	file:///Users/almog.gavra/dev/confluent/ksql/ksqldb-functional-tests/target/test-classes/sql-tests/test.sql:21
```

### PR DESCRIPTION
**Pull request by jzaralim**
### Description 
Make the assert executor only check the values of the columns specified in the assert command instead of the entire record.
Fixes #6056

### Testing done 
Added a YATT test.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Review by vpapavas — COMMENTED** [review]
Thank you @jzaralim ! LGTM with one question: is the column `ROWOFFSET` accessible by the user? I can't remember atm. If yes, would be good to add a check for it in the method and a test case for it as well.

**Comment by jzaralim** [comment]
@vpapavas In normal ksql it is accesible, but not in YATT. Making YATT work well with metadata columns is something we're planning though.

File: ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/driver/AssertExecutor.java [hunk 1]
```diff
@@ -217,6 +213,33 @@ private static void assertContent(
...
 
+  private static boolean verifyValues(
+      final KsqlGenericRecord expected,
+      final KsqlGenericRecord actual,
+      final DataSource dataSource,
+      final List<ColumnName> columns
+  ) {
+    final LogicalSchema schema = dataSource.getSchema().withPseudoAndKeyColsInValue(false);
+    for (final ColumnName col : columns) {
+      if (schema.isKeyColumn(col)) {
```
Comment (bvarghese1) [inline ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/driver/AssertExecutor.java:224]: For key schema do we also want to check if the no of fields in actual and expected are the same?

Comment (jzaralim) [reply ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/driver/AssertExecutor.java:229]: Added this check for key and value. Also added some tests for asserting tombstones.

**Review by vpapavas — APPROVED** [review]
LGTM! Thank you @jzaralim!

**Pull request merged**